### PR TITLE
Docs: Hutch::Config

### PR DESCRIPTION
**Pull request by olleolleolle**
In order to help users configure Hutch, try to tell the configuration story using YARD docs. This PR now produces docs.

screenshot 2016-04-29 06 55 13

Using a YARD processor, the names, types and default values can be presented in the documentation:

screenshot 2016-04-30 15 14 21

(Using the power of a CSS class, we could style that to be legible.)

### CODE REVIEW

File: lib/hutch/config.rb [hunk 1]
```diff
@@ -5,42 +5,75 @@
...
-                   daemonise
-                   publisher_confirms
-                   force_publisher_confirms
-                   enable_http_api_use
-                   consumer_pool_abort_on_exception).freeze
-
-    ALL_KEYS = (BOOL_KEYS + NUMBER_KEYS + STRING_KEYS).map(&:to_sym).freeze
+    STRING_KEYS = Set.new
+    NUMBER_KEYS = Set.new
+    BOOL_KEYS = Set.new
```
Comment (olleolleolle) [inline lib/hutch/config.rb]: TODO: Make these not be defined as constants.

File: lib/hutch/config.rb [hunk 2]
```diff
@@ -5,42 +5,75 @@
...
+
+    boolean_setting :mq_tls, false
+    boolean_setting :mq_verify_peer, true
+    boolean_setting :mq_api_ssl, false
+    boolean_setting :autoload_rails, true
+    boolean_setting :daemonise, false
+    boolean_setting :publisher_confirms, false
+    boolean_setting :force_publisher_confirms, false
+    boolean_setting :enable_http_api_use, true
+    boolean_setting :consumer_pool_abort_on_exception, false
```
Comment (olleolleolle) [inline lib/hutch/config.rb]: Perhaps these DSL-like methods could help set up the default_settings in some way.

File: lib/yard-settings/handler.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  handles method_call :string_setting
+  handles method_call :number_setting
+  handles method_call :boolean_setting
+
+  namespace_only
+
+  def process
+    name = statement.parameters.first.jump(:tstring_content, :ident).source
+    object = YARD::CodeObjects::MethodObject.new(namespace, name)
+    register(object)
```
Comment (olleolleolle) [inline lib/yard-settings/handler.rb]: This is the "render each of the settings as a method on the Module" bit. Perhaps this is over-reaching, as the non-ENV-settable settings (which are also available as method calls via method_missing) will not be represented here.

Example of what the current output is:

screenshot 2016-04-30 15 18 43

File: templates/default/layout/html/settings.erb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    <tr>
+    <th><tt><%= task.full_path %></tt></th>
+    <% if task.full_path == 'default' %>
+    <th><%= task.tag(:task_deps).text %></th>
+    <% else %>
+    <td><%= task.docstring %></td>
+    <% end %>
+    </tr>
+  <% end %>
+</table>
```
Comment (olleolleolle) [inline templates/default/layout/html/settings.erb]: This file should go.

**Comment by michaelklishin** [comment]
This is awesome, thank you @olleolleolle! I'd be quite happy to merge the same kind of PR for Bunny ;)

**Comment by olleolleolle** [comment]
@michaelklishin Thanks for merging - I believe there are a few clean-ups that can be achieved. I'll be PRing on that account soon.

File: lib/yard-settings/handler.rb (hunk 3 above)
Comment (olleolleolle) [inline lib/yard-settings/handler.rb]: @michaelklishin This is the place where I'd like your input on also creating "`read_timeout=`" as well (perhaps with the "Found using `method_missing`" addition, too?

**Pull request merged**